test/shopify_app/login_protection_test.rb

```diff
@@ -7,6 +7,7 @@ class LoginProtectionController < ActionController::Base
   helper_method :shop_session
 
   before_action :login_again_if_different_shop, only: [:second_login]
+  around_action :shopify_session, only: [:with_a_shopify_session]
```

Comment:
Good suggestion! I was able to throw `#shopify_session` around the `index` action.